src/posix/platform/multicast_routing.cpp

```diff
@@ -309,6 +309,8 @@ otError MulticastRoutingManager::AddMulticastForwardingCache(const Ip6::Address
     }
     else
     {
+        VerifyOrExit(otIp6PrefixMatch(&aSrcAddr, otThreadGetMeshLocalEid(gInstance)) < OT_IP6_PREFIX_BITSIZE,
+                     error = OT_ERROR_INVALID_ARGS);
```

Comment:
Returning OT_ERROR_INVALID_ARGS would cause a warning message to be printed. 

I think we should avoid the warnings?

Reply:
I agree. Shall I use `OT_ERROR_NONE` here?

Reply:
After thinking twice, I think it's still a valid warning. Is there any concern about having this as a warning?

Reply:
It's not a valid warning?
User applications may send multicast packets with MLEID source to reach all nodes within the Thread network?